src/es/olympusscans/src/eu/kanade/tachiyomi/extension/es/olympusscans/olympusscans.kt

```diff
@@ -0,0 +1,149 @@
+        val jsonTemp = Json { ignoreUnknownKeys = true }
+        val resultPayload = jsonTemp.decodeFromString<PayloadMangaDto>(response.body.string())
+        val mangas = resultPayload.data.map {
+            SManga.create().apply {
+                setUrlWithoutDomain("$baseURLApi/${it.slug}")
+                title = it.name
+                thumbnail_url = it.cover
+            }
+        }
+        return MangasPage(mangas, hasNextPage = false)
+    }
+
+    override fun searchMangaSelector(): String =
+        throw UnsupportedOperationException("Not used.")
+
+    override fun searchMangaFromElement(element: Element): SManga =
+        throw UnsupportedOperationException("Not used.")
+
+    override fun searchMangaNextPageSelector(): String? =
+        null
+
+    // Latest
+    override fun latestUpdatesRequest(page: Int): Request = throw UnsupportedOperationException("Not used.")
+
+    override fun latestUpdatesSelector(): String =
+        ".mainpage-manga"
+
+    override fun latestUpdatesFromElement(element: Element): SManga = throw Exception("Not used")
+
+    override fun latestUpdatesNextPageSelector(): String? =
+        null
+
+    // Details
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        title = document.selectFirst("h1")!!.text()
+    }
+
+    // Chapters
+
+    override fun chapterListSelector() = "not using"
+
+    override fun chapterFromElement(element: Element): SChapter = throw UnsupportedOperationException("Not used")
+
+    override fun chapterListRequest(manga: SManga): Request {
+        return paginatedChapterListRequest(manga.url, 1)
+    }
+
+    private fun paginatedChapterListRequest(mangaUrl: String, page: Int): Request {
+        return GET(
+            url = "$baseURLApi/api/series$mangaUrl/chapters?page=$page&direction=desc&type=comic",
+            headers = headers,
+        )
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val slug = response.request.url.toString().substringAfter("/api/series/").substringBefore("/chapters")
+        var data = json.decodeFromString<PayloadChapterDto>(response.body.string())
+        var resultSize = data.data.size
+        var page = 2
+        while (data.meta.total > resultSize) {
+            val newRequest = paginatedChapterListRequest("/$slug", page)
+            val newResponse = client.newCall(newRequest).execute()
+            var newData = json.decodeFromString<PayloadChapterDto>(newResponse.body.string())
+            data.data += newData.data
+            resultSize += newData.data.size
+            page += 1
+        }
+        return data.data.map { chap -> chapterFromObject(chap, slug) }
+    }
+
+    private fun chapterFromObject(chapter: ChapterDto, slug: String) = SChapter.create().apply {
+        url = "/capitulo/${chapter.id}/comic-$slug"
+        name = "Capitulo ${chapter.name}"
+        chapter_number = chapter.name!!.toFloat()
```

Comment:
What if it suddenly isn't a valid float any more? You need to handle that as well.

```suggestion
        chapter_number = chapter.name.toFloatOrNull() ?: -1f
```

Do fill the `upload_date` as well. There's some examples in the CONTRIBUTING.md guide.